Server/ReasnAPI/ReasnAPI/Services/AddressService.cs

```diff
@@ -1,16 +1,83 @@
-﻿using ReasnAPI.Models.Database;
-
-namespace ReasnAPI.Services {
-    public class AddressService (ReasnContext context) {
-        private readonly ReasnContext _context = context;
-
-        /* TODO: Create following functions for this class
-         * create
-         * update
-         * delete
-         * get by ID
-         * get list by filter
-         * get all
-         */
-    }
-}
+using ReasnAPI.Mappers;
+using ReasnAPI.Models.Database;
+using ReasnAPI.Models.DTOs;
+using ReasnAPI.Services.Exceptions;
+using System.Linq.Expressions;
+
+namespace ReasnAPI.Services;
+
+public class AddressService(ReasnContext context)
+{
+    public AddressDto? CreateAddress(AddressDto? addressDto)
+    {
+        ArgumentNullException.ThrowIfNull(addressDto);
+
+        context.Addresses.Add(addressDto.ToEntity());
+        context.SaveChanges();
+
+        return addressDto;
+    }
+
+    public AddressDto? UpdateAddress(int addressId, AddressDto? addressDto)

```

Comment:
Same here, you're throwing execption when resource is not found - the resulting object will exist. Please, remove "?" where needed in the rest of the code.